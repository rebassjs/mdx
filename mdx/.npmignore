docs
src
.babelrc
babel.config.js
test
coverage
