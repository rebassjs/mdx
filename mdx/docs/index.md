import { Helmet } from 'react-helmet'
import {
  RebassMDX,
  Styled,
  Box,
} from '../src'
import Readme from '../README.md'

export default props =>
  <RebassMDX wrap>
    {props.children}
  </RebassMDX>

<Helmet>
  <title>@rebass/mdx</title>
</Helmet>

<Readme />

---

## Usage

```jsx
import React from 'react'
import RebassMDX from '@rebass/mdx'
import Document from './document.mdx

export default props =>
  <RebassMDX>
    <Document />
  </RebassMDX>
```

### Props

- `theme` add a custom [Rebass theme][]
  - `theme.styles` override component styles
- `components` override specific components or add custom components to code fence examples

[Rebass theme]: https://rebassjs.org/theming
