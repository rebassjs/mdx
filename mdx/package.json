{
  "name": "@rebass/mdx",
  "version": "2.0.0-0",
  "description": "MDX Provider for styled components",
  "main": "dist/index.js",
  "scripts": {
    "prepare": "babel src -d dist",
    "start": "gx docs"
  },
  "keywords": [],
  "author": "Brent Jackson",
  "license": "MIT",
  "dependencies": {
    "@emotion/core": "^10.0.10",
    "@emotion/styled": "^10.0.10",
    "@mdx-js/mdx": "^1.0.0-alpha.10",
    "@mdx-js/tag": "^1.0.0-alpha.6",
    "@styled-system/css": "^1.0.2",
    "emotion-mdx": "^1.0.0-4",
    "emotion-theming": "^10.0.10",
    "lodash.merge": "^4.6.1",
    "styled-system": "^4.1.0"
  },
  "devDependencies": {
    "@babel/cli": "^7.2.3",
    "@babel/core": "^7.4.0",
    "@babel/preset-env": "^7.4.2",
    "@babel/preset-react": "^7.0.0",
    "@rebass/emotion": "^3.1.0",
    "rebass": "^3.1.0",
    "gx": "^2.0.0-20",
    "react-syntax-highlighter": "^10.2.1"
  }
}
