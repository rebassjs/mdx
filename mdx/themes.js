module.exports = require('./dist/themes')
