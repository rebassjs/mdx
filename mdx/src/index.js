import React from 'react'
import {
  ComponentProvider,
  Styled
} from 'emotion-mdx'
import css from '@styled-system/css'
import * as Rebass from '@rebass/emotion'
import { base } from './themes'

export { useComponents, Styled } from 'emotion-mdx'

export const components = {
  ...Rebass,
}

export const RebassMDX = ({
  wrap,
  ...props
}) => {
  const children = wrap ? (
    <Styled.wrapper>
      {props.children}
    </Styled.wrapper>
  ) : props.children

  return (
    <ComponentProvider
      {...props}
      transform={css}
      children={children}
    />
  )
}

RebassMDX.defaultProps = {
  components,
  theme: base,
}

RebassMDX.props = defaults => props =>
  <RebassMDX
    {...defaults}
    {...props}
  />

const comp = tag => React.forwardRef((props, ref) =>
  <Styled
    {...props}
    ref={ref}
    tag={tag}
  />
)

export const Box = comp('Box')
export const Flex = comp('Flex')
export const Text = comp('Text')
export const Heading = comp('Heading')
export const Button = comp('Button')
export const Link = comp('Link')
export const Image = comp('Image')
export const Card = comp('Card')

export default RebassMDX
