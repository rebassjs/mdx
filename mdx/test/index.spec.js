import React from 'react'
import mdx from '@mdx-js/mdx/create-element'
import { useMDXComponents } from '@mdx-js/react'
import renderer from 'react-test-renderer'
import { matchers } from 'jest-emotion'
import RebassMDX from '../src'

expect.extend(matchers)

const Hello = props => {
  const { h1 } = useMDXComponents()
  return mdx(h1, props, 'Hello')
}

test('renders', () => {
  const json = renderer.create(
    mdx(RebassMDX, null,
      mdx(Hello)
    )
  ).toJSON()
  expect(json).toHaveStyleRule('font-size', '32px')
  expect(json).toHaveStyleRule('font-family', 'inherit')
  expect(json).toHaveStyleRule('font-weight', '700')
  expect(json).toMatchSnapshot()
})

test('renders with wrapper', () => {
  const json = renderer.create(
    mdx(RebassMDX, { wrap: true },
      mdx(Hello)
    )
  ).toJSON()
  expect(json.type).toBe('div')
  expect(json).toMatchSnapshot()
})
