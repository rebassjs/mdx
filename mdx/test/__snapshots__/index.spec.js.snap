// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
.emotion-0 {
  margin-top: 32px;
  margin-bottom: 32px;
  font-family: inherit;
  font-size: 32px;
  font-weight: 700;
  line-height: 1.25;
}

.emotion-0 a {
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0 a:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

@media screen and (min-width:40em) {
  .emotion-0 {
    font-size: 48px;
  }
}

<h1
  className="emotion-0"
>
  Hello
</h1>
`;

exports[`renders with wrapper 1`] = `
.emotion-0 {
  margin-top: 32px;
  margin-bottom: 32px;
  font-family: inherit;
  font-size: 32px;
  font-weight: 700;
  line-height: 1.25;
}

.emotion-0 a {
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0 a:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

@media screen and (min-width:40em) {
  .emotion-0 {
    font-size: 48px;
  }
}

.emotion-1 {
  color: #000;
  background-color: #fff;
  font-family: system-ui,sans-serif;
  font-weight: 400;
  line-height: 1.5;
  color: #000;
  background-color: #fff;
  font-family: system-ui,sans-serif;
  font-weight: 400;
  line-height: 1.5;
}

<div
  className="emotion-1"
>
  <h1
    className="emotion-0"
  >
    Hello
  </h1>
</div>
`;
